cmd/prometheus/main.go

```diff
@@ -461,7 +463,7 @@ func main() {
 	a.Flag("scrape.discovery-reload-interval", "Interval used by scrape manager to throttle target groups updates.").
 		Hidden().Default("5s").SetValue(&cfg.scrape.DiscoveryReloadInterval)
 
-	a.Flag("enable-feature", "Comma separated feature names to enable. Valid options: agent, auto-gomemlimit, exemplar-storage, expand-external-labels, memory-snapshot-on-shutdown, promql-per-step-stats, promql-experimental-functions, extra-scrape-metrics, new-service-discovery-manager, auto-gomaxprocs, no-default-scrape-port, native-histograms, otlp-write-receiver, created-timestamp-zero-ingestion, concurrent-rule-eval. See https://prometheus.io/docs/prometheus/latest/feature_flags/ for more details.").
+	a.Flag("enable-feature", "Comma separated feature names to enable. Valid options: agent, auto-gomemlimit, exemplar-storage, expand-external-labels, memory-snapshot-on-shutdown, promql-per-step-stats, promql-experimental-functions, extra-scrape-metrics, new-service-discovery-manager, auto-gomaxprocs, no-default-scrape-port, native-histograms, otlp-write-receiver, created-timestamp-zero-ingestion, concurrent-rule-eval, legacy-names. See https://prometheus.io/docs/prometheus/latest/feature_flags/ for more details.").
```

Comment:
`enable-feature` is for experimental features. `legacy-names` is neither breaking nor experimental. I think this should be a flag of its own or something else. Another argument can be made that in the future we want to always have utf8 enabled in Prometheus and remove the `legacy-names`, in which case feature flags are better. @beorn7 WDYT?

Reply:
See the [below comment](https://github.com/prometheus/prometheus/pull/14705#pullrequestreview-2257691818); we figured we don't need the flag afterall